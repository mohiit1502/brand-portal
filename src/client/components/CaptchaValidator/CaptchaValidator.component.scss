@import "../../styles/override/variables";
.c-CaptchaValidator {
    width: fit-content;
    height: fit-content;
    border: 1px;
    border-color: rgb(168, 164, 164);
    border-style: solid;
    .imageContainer {
      border: 1px;
      border-color: rgb(121, 116, 116);
      border-style: solid;

      .refresh {
          height: 20px;
          width: 20px;
          text-align: right;
      }
  }

  .btn {
    &.btn-primary {
        background: $market-place-blue !important;
        border: 1px solid;
        opacity: 1;
    }
    }

}
