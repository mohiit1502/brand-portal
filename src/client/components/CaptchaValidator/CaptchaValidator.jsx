import React, {useEffect, useState} from "react";
import PropTypes from "prop-types";
import Http from "../../utility/Http";
import ReCAPTCHA from "react-google-recaptcha";
import "./CaptchaValidator.component.scss";

const CaptchaValidator = props => {
  const [captchaConfig, setCaptchaConfig] = useState();
  const [isValid, setValid] = useState(true);
  useEffect(() => {
    !captchaConfig  &&
    Http.get("/api/getCaptchaConfig")
    .then(res => {
      setCaptchaConfig(res.body);
      if(!res.body.enableCaptcha) {
        verifyCaptcha(true);
      }
    }).catch(e => {
      setCaptchaConfig({});
      verifyCaptcha(true);
      console.log(e)
    });
  }, [captchaConfig]);

  const verifyCaptcha = res => {
    if (res) {
      setValid(true);
      const dummyCaptchaEvent = {
        target: {
          value: true
        }
      };
      props.onChange && props.onChange(dummyCaptchaEvent, props.inputId);
    }
  };
  const onExpired = res => {
    setValid(false);
    const dummyCaptchaEvent = {
      target: {
        value: false
      }
    };
    props.onChange && props.onChange(dummyCaptchaEvent, props.inputId);
  };

  return (
    <div className={`c-CaptchaValidator mx-auto ${!captchaConfig ? " captcha-container loader" : ""}`}>
      {
        captchaConfig && captchaConfig.enableCaptcha &&
            <React.Fragment>
              <ReCAPTCHA sitekey={captchaConfig.sitekey} onChange={verifyCaptcha} onExpired={onExpired}/>
              <small className={`form-text custom-input-help-text text-danger text-center`}>
                {props.error}
              </small>
            </React.Fragment>
      }
    </div>
  );
};

CaptchaValidator.propTypes = {
  captchaFieldName: PropTypes.string,
  error: PropTypes.string,
  onSubmit: PropTypes.func,
  submitButtonLabel: PropTypes.string
};

export default CaptchaValidator;
