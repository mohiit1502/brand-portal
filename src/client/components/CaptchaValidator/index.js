import CaptchaValidator from "./CaptchaValidator.jsx";

export default CaptchaValidator;
