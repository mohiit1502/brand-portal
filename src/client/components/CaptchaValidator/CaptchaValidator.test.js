/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React from "react";
import CaptchaValidator from "./CaptchaValidator";

describe("CaptchaValidator", () => {
   it("renders without error", () => {

   });
});
