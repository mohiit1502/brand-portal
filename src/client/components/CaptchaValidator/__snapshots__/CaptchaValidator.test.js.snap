// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaptchaValidator test container CaptchaValidator renders without error should render the Captcha successfully when enabled 1`] = `
<div
  className="c-CaptchaValidator  captcha-container loader"
/>
`;

exports[`CaptchaValidator test container CaptchaValidator renders without error tests API failure scenario 1`] = `
<div
  className="c-CaptchaValidator  captcha-container loader"
/>
`;

exports[`CaptchaValidator test container CaptchaValidator renders without error verify Captcha when disabled 1`] = `
<div
  className="c-CaptchaValidator  captcha-container loader"
/>
`;
